docs/rfcs/xxxx-backwards-compatibility.md

```diff
@@ -0,0 +1,176 @@
+# Grafana Agent backwards compatibility 
+
+* Date: 2023-05-25
+* Author: Robert Fratto (@rfratto)
+* PR: [grafana/agent#3981](https://github.com/grafana/agent/pull/3981)
+* Status: Draft
+
+Grafana Agent has been following [semantic versioning](https://semver.org/)
+since its inception. After three years of development and 33 minor releases,
+the project is on trajectory to have a 1.0 release. 
+
+In the context of semantic versioning, a 1.0 release indicates that future
+minor releases have backwards compatibility with older minor releases in the
+same major release; version 1.1 is backwards compatible with version 1.0.
+Having major releases signals to users when they can upgrade with more
+confidence (minor releases) and when an upgrade may take more time (new major
+releases). 
+
+However, Grafana Agent is a large project, with a large surface area for what
+may be considered part of the backwards compatibility guarantees. This proposal
+formally establishes what parts of Grafana Agent will be protected by backwards
+compatibility. 
+
+## Goals 
+
+- Set expectations for what is covered by backwards compatibility. 
+- Set expectations for when upgrades to new major versions will be forced.
+
+## Proposal 
+
+Backwards compatibility means that a user can upgrade their version of Grafana
+Agent without needing to make any changes with the way they interact with
+Grafana Agent. 
+
+### Scope of backwards compatibility   
+
+The following will be protected by backwards compatibility between minor
+releases: 
+
+- **User configuration**, including configuration files and the command-line
+  interface. 
+
+- **Versioned network APIs**, if any versioned APIs are introduced prior to the
+  1.0 release.
+
+- **Telemetry data used in official dashboards**. This means that users will
```

Comment:
Can we define official? Is it any dashboard contained inside the Agent Repo or any dashboard used in a grafana blessed mixin?